### PR TITLE
[Container] Fix colorScheme and typographyScheme to be nonnull.

### PR DESCRIPTION
**Pull request by codeman7**
## Introduction
Before we move container scheme to ready we need to have both the `colorScheme` and `typographyScheme` nonnull. By making colorScheme and typographyScheme of MDCContainerScheming nonnull, clients will reduce the amount of conditional checks required in their apps and reduce a category of potential errors (accidentally passing nil to a nonnull parameter).

## The problem
Both `colorScheme` and `typographyScheme` were nullable in the container scheme making it harder for clients to use the container scheme throughout their app on non Material UIElements, i.e. UIView.backgroundColor

## The fix
Make both of these properties nonnull.

## Additional work
- [x]  I had to update the swift examples to not be optional and I additionally removed the nil checks from the ObjC examples.

### CODE REVIEW

**Review by yarneo — APPROVED** [review]

File: components/ActionSheet/examples/ActionSheetComparisonExampleViewController.m [hunk 1]
```diff
@@ -46,15 +46,7 @@ - (instancetype)init {
 - (void)viewDidLoad {
   [super viewDidLoad];
 
-  id<MDCColorScheming> colorScheme;
```
Comment (romoore) [inline components/ActionSheet/examples/ActionSheetComparisonExampleViewController.m]: If this code was generating a compiler error, then I think it's good to change it. If not, I think it should be moved to a separate PR and this should focus only on the critical path to make changes to MDCContainerScheming.

File: components/Cards/examples/EditReorderCollectionViewController.swift [hunk 2]
```diff
@@ -34,11 +34,11 @@ class EditReorderCollectionViewController: UIViewController,
   var containerScheme: MDCContainerScheming
 
   var colorScheme: MDCColorScheming {
```
Comment (romoore) [inline components/Cards/examples/EditReorderCollectionViewController.swift]: Seems like these two computed vars can be removed entirely and their use replaced with `containerScheme.colorScheme`.

File: components/schemes/Container/src/MDCContainerScheme.h [hunk 3]
```diff
@@ -49,8 +49,8 @@
 __attribute__((objc_subclassing_restricted)) @interface MDCContainerScheme
```
Comment (romoore) [inline components/schemes/Container/src/MDCContainerScheme.h:47]: Please be sure to update the header documentation now that properties are no longer `nil`.  The defaults should be explicitly documented.

**Review by romoore — DISMISSED** [review]

File: components/ActionSheet/examples/ActionSheetTypicalUseExampleViewController.m [hunk 4]
```diff
@@ -44,15 +44,7 @@ - (instancetype)init {
...
-  id<MDCColorScheming> colorScheme;
-  if (self.containerScheme.colorScheme != nil) {
-    colorScheme = self.containerScheme.colorScheme;
-  } else {
-    colorScheme =
-        [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
-  }
-
-  self.view.backgroundColor = colorScheme.backgroundColor;
+  self.view.backgroundColor = self.containerScheme.colorScheme.backgroundColor;
```
Comment (galiak11) [inline components/ActionSheet/examples/ActionSheetTypicalUseExampleViewController.m]: I think checking if self.containerScheme is nil is still required (and assigning it a default container-scheme if it's nil)), because it may not be injected.
This is true for all examples.

File: components/schemes/Container/src/MDCContainerScheme.h [hunk 5]
```diff
@@ -26,12 +26,12 @@
 /**
  The color scheme to use for the container scheme.
  */
-@property(nonatomic, nullable, readonly) id<MDCColorScheming> colorScheme;
+@property(nonatomic, nonnull, readonly) id<MDCColorScheming> colorScheme;
```
Comment (galiak11) [inline components/schemes/Container/src/MDCContainerScheme.h:29]: Should we provide an initializer now that these are not null? 
How are we expecting client apps to assign these values on a newly initialized container scheme?

Comment (romoore) [reply components/schemes/Container/src/MDCContainerScheme.h:29]: @galiak11  There is an `init` method for `MDCContainerScheme` in the `.m` file. I don't think there's much value dictating initialization/construction in the protocol - implementations may use builders, factories, `init` methods, etc.   The protocol is focused on consumption - requiring the properties, their types and attributes.  

Clients can assign the properties directly to the implementation properties.

File: components/schemes/Container/src/MDCContainerScheme.h (hunk 3 above)
Comment (codeman7) [reply components/schemes/Container/src/MDCContainerScheme.h:47]: Done

File: components/ActionSheet/examples/ActionSheetComparisonExampleViewController.m (hunk 1 above)
Comment (codeman7) [reply components/ActionSheet/examples/ActionSheetComparisonExampleViewController.m]: Removed

**Review by randallli — APPROVED** [review]

File: components/schemes/Container/src/MDCContainerScheme.h (hunk 5 above)
Comment (codeman7) [reply components/schemes/Container/src/MDCContainerScheme.h:29]: I think we just intend to have the `init` in the `.m` file as @romoore has outlined. Do you agree with that?

Comment (galiak11) [reply components/schemes/Container/src/MDCContainerScheme.h:29]: Yes.

File: components/Cards/examples/EditReorderCollectionViewController.swift (hunk 2 above)
Comment (codeman7) [reply components/Cards/examples/EditReorderCollectionViewController.swift]: These have been updated.

**Review by galiak11 — APPROVED** [review]

File: components/schemes/Container/src/MDCContainerScheme.h (hunk 3 above)
Comment (romoore) [reply components/schemes/Container/src/MDCContainerScheme.h:47]: This is still not done - please fix the header documentation above.

**Review by romoore — APPROVED** [review]
Please fix the header comment before merging.

File: components/schemes/Container/src/MDCContainerScheme.h (hunk 3 above)
Comment (codeman7) [reply components/schemes/Container/src/MDCContainerScheme.h:47]: Good catch, I didn't see that. It has been fixed.

**Comment by codeman7** [comment]
This has been tested in cl/235759497

File: components/ActionSheet/examples/ActionSheetTypicalUseExampleViewController.m (hunk 4 above)
Comment (codeman7) [reply components/ActionSheet/examples/ActionSheetTypicalUseExampleViewController.m]: This has been updated.
